### PR TITLE
Fix format printing for non-address targets

### PR DESCRIPTION
**Pull request by Alexeyan**
Regarding Issue https://github.com/angr/angr/issues/1727 

This patch fixes the issue for me.
But since i'm not sure which types the src and dst can have, maybe @ltfish should take a look at this and if it's not sufficient i can add the missing type formats.

Best regards,
Alex

### CODE REVIEW

File: angr/knowledge_plugins/xrefs/xref.py [hunk 1]
```diff
@@ -34,9 +34,9 @@ def type_string(self):
 
     def __repr__(self):
         return "<XRef %s: %s->%s>" % (
-                XRefType.to_string(self.type),
+                self.type_string,
                 "%#x" % self.ins_addr if self.ins_addr is not None else "%#x[%d]" % (self.block_addr, self.stmt_idx),
-                "%#x" % (self.dst if self.dst is not None else self.memory_data.addr)
+                "%#s" % self.dst if self.dst is not None else "%#x" % (self.memory_data.addr)
```
Comment (ltfish) [inline angr/knowledge_plugins/xrefs/xref.py]: What's the difference between `%#s` and `%s`?

Comment (Alexeyan) [reply angr/knowledge_plugins/xrefs/xref.py]: Nothing. `#`s alternate form isn't specified for strings. I missed it.

**Comment by rhelmot** [comment]
as a side note - and I don't know if this is necessary for right now, but something to be mindful of - I think we should switch to a formatting mechanism which invokes the `__format__` magic method so we can just define that for alternate address types. For example `"{:x}".format(foo)` will invoke something along the lines of `foo.__format__("x")`

**Comment by Alexeyan** [comment]
I would say this can be done incremental. 
Is anything missing for the current pr right now?

**Comment by rhelmot** [comment]
nothing is missing. lgtm

**Comment by ltfish** [comment]
Merged. Thanks!

**Pull request merged**